modules/drivers/bigquery-cloud-sdk/test/metabase/test/data/bigquery_cloud_sdk.clj

```diff
@@ -329,18 +329,19 @@
 (defn- load-tabledef! [dataset-id {:keys [table-name field-definitions], :as tabledef}]
   (let [table-name (normalize-name table-name)]
     (create-table! dataset-id table-name field-definitions)
-    ;; retry the `insert-data!` step up to 5 times because it seens to fail silently a lot. Since each row is given a
-    ;; unique key it shouldn't result in duplicates.
-    (loop [num-retries 5]
-      (let [^Throwable e (try
-                           (insert-data! dataset-id table-name (tabledef->prepared-rows tabledef))
-                           nil
-                           (catch Throwable e
-                             e))]
-        (when e
-          (if (pos? num-retries)
-            (recur (dec num-retries))
-            (throw e)))))))
+    (when (seq (:rows tabledef))
```

Comment:
Just curious - is this just an optimization, or would the `insert-data!` call fail?

Reply:
I think it failed during testing, because I test all drivers with empty datasets (schema only) which is new at metabase so a lot of these kinds of fns crashed for empty inputs.

Now one could make an argument for crashing being the correct behaviour -but I would have had to change more code (at least at a glance!)